edb/lang/edgeql/compiler/schemactx.py

```diff
@@ -40,12 +40,16 @@
 
 
 def get_schema_object(
-        name: typing.Union[str, qlast.ObjectRef],
+        name: typing.Union[str, qlast.ObjectRef, qlast.TypeName],
         module: typing.Optional[str]=None, *,
         item_types: typing.Optional[typing.List[s_obj.ObjectMeta]],
         ctx: context.ContextLevel,
         srcctx: typing.Optional[parsing.ParserContext] = None) -> s_obj.Object:
 
+    if isinstance(name, qlast.TypeName):
```

Comment:
It would be better to do this at call site, where there is more context as to what is going on.

Reply:
OK, I'm moving this bit of logic out to the call sites.